### PR TITLE
Bxmsdoc 2576 Case modeling content

### PR DESCRIPTION
**Pull request by gsheldon**
case modeling content from Getting started with case management story.

### CODE REVIEW

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [id='case-management-designing-a-case-proc']
```
Comment (sterobin) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc:1]: Not shown in this PR, so recording these here:

Preface
- "using this project using the Showcase application" - Maybe reword. I'd propose an alternative, but I'm not sure what you actually mean.
- Prereqs: First two bullets should be "... is installed" for consistency.

Ch 1 Case management: 
- "end to end flow" > "end-to-end flow"
- "real world applications" > "real-world applications"
- "A Per case runtime strategy" > "A per-case runtime strategy" (technically, but if it's too ugly, at least just lc, with no hypen if you'd like)

File: docs/product-assembly_designing-and-building-cases/src/main/asciidoc/case-management-case-modeling-proc.adoc [hunk 2]
```diff
@@ -1,2 +1,36 @@
...
-= TODO
\ No newline at end of file
+[id='assembly_case-management-case-modeling-{context}']
+= Modeling a case definition
+
+A case definition represents the flow of the case management project that differs from a normal business process workflow in that it has no clear start node. A case is event-driven with ad hoc interaction between tasks and processes, rather than following the sequential flow of a business process. Case definitions can include nodes that are configured to start automatically when a new case instance is created, as well as ad hoc nodes that are optional and can be triggered during a case when required.
+
+Cases are modeled using the process design tool, which you can access by opening your project in {CENTRAL} and clicking *Add Asset* -> *Case definition*.
+
+The following case management features can be included and configured in a case definition:
```
Comment (sterobin) [inline docs/product-assembly_designing-and-building-cases/src/main/asciidoc/case-management-case-modeling-proc.adoc:8]: Switch Case roles to second in line and Ad hoc to first, to be consistent with how they actually appear in the doc. And nice work with the ifevals! Those can be tedious to set up correctly.

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc [hunk 3]
```diff
@@ -65,8 +72,14 @@ image::case_roles.png[Case Roles]
 * `manager` - the employee's manager; the person who will approve or deny the requested hardware. The role cardinality is set to `1`, which means that only one person or group can be assigned to this role.
 * `supplier` - the available suppliers of IT hardware in the system. There is usually more than one supplier.
```
Comment (sterobin) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc]: These bulleted descriptions should use a colon followed by capital, "owner: The employee ...". This is consistent with the rest of your doc and with our general style throughout the docs.

Also in this chapter but not in this PR: "process level and case level variables" > "process-level and case-level variables (again, technically, but your call)

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc [hunk 4]
```diff
@@ -89,7 +102,14 @@ In the IT Orders case, the status of `Milestone 1: Order placed` depends on the
...
-* For more information about the milestones in the IT Orders case, see <<case-management-itorders-milestones-ref-{context}>>. 
+* For more information about the milestones in the IT Orders case, see 
+ifdef::case-management-getting-started[]
+<<case-management-itorders-milestones-ref-{context}>>.
+endif::[]
+ifdef::case-management-design[]
+{URL_GETTING_STARTED_CASES}[_{GETTING_STARTED_CASES}_].
+endif::[]
+
 * To learn more about milestones, see <<case-management-milestones-con-{context}>>.
```
Comment (sterobin) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc]: Above, you have a step "Add a milestone to the case ..." followed by a single sub-step. Not sure if that was intentional, but typically single steps should be avoided, consolidated, or not numbered. Should you just make it the next step? I don't know enough about the process to say. Or it can just be an indented part of the super-step or something.

And you have the "For more info" bits as bullets. I'd nix the bullets and just left align with the numbered indentation.

Also, the milestones cross-refs are not showing up in the rendered output because they're not set up right. I assume you meant to use ifevals?

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc [hunk 5]
```diff
@@ -50,7 +50,14 @@ image::case-variables.png[Editor for IT Orders variables]
...
 +
-For more information about the case file, see <<case-management-case-file-con-{context}>>.
+For more information about the case file, see 
+ifeval::["{context}" == "case-management-getting-started"]
+<<case-management-case-file-con-case-management-getting-started>>
+endif::[]
+ifeval::["{context}" == "case-management-design"]
+<<case-management-case-file-con-case-management-design>>
+endif::[]
+.
```
Comment (sterobin) [inline docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc:60]: General comment about cross-refs throughout the doc.

The majority of the time you use the ref format "<<anchor-id-context-A>>", "<<anchor-id-context-B>>", etc., and occasionally "xref:anchor-id-context-A", etc. We decided back when we started modularizing that we would start using only the xref: format (Contrib Guide) because the << >> format starts to break down in certain cases of cross-referencing reused modules. I'm surprised you were able to use the ifevals using the << >> way! I found that it only worked with xref and hit issues with the << >> when referencing the same module in multiple places. So that's cool if it works. There was one other reason though that I can't recall. It's up to you I guess. So long as it works the same, I suppose it's all that really matters.

Also, this is super, SUPER trivial and I honestly don't care if you heed it or not, just informing. The actual format for setting anchor IDs with a context is to use hyphens all around but underscore before the context attribute: "[id='anchor-id_{context}']". This would make the ref the following, for example: `xref:case-management-case-file-con_case-management-design`. Makes zero difference to the function, but makes it easier to delimit the anchor ID from the context variable in cross-refs, in the rendered doc's URL, etc., which is why we went with that. Section about this in the contrib guide: http://ccs-jenkins.gsslab.brq.redhat.com:8080/view/Topic%20branches/job/doc-Red_Hat_JBoss_BxMS_Contribution_Guide-branch-shared-repo/lastSuccessfulBuild/artifact/index.html#reusing-modules

Again, completely up to you if you want to ignore this. I know it's one of many, many things around here that are error prone and arbitrary. Just wanted to at least bring it up.

**Review by sterobin — COMMENTED** [review]
@gsheldon, done. Mostly minor comments. Please don't hate me :) As usual, feel free to ignore anything you'd like.

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc (hunk 4 above)
Comment (gsheldon) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc]: Ah yes, I meant to use ifevals!

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc (hunk 1 above)
Comment (gsheldon) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc:1]: * The Showcase sentence was a typo. Fixed.
* I feel like I already fixed those hyphenations before in a previous peer review but for some reason they haven't been saved. Very strange, but they're fixed now, hopefully the change sticks.
* Per case is what it actually is in the UI, but I'll chuck some `` around it to make that more obvious.

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc (hunk 5 above)
Comment (gsheldon) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc:60]: I'm not sure when the discussion around <> vs xref tags happened, but I started doing the <<>> ones when we were putting out LA because I saw that's how everyone else had been doing internal links. I've been doing them that way ever since. 

I'm not able to change all of those links at this point and as you said, if it's working then that's all that matters for now. If possible, it'd be good to have changes communicated through email or something because I don't check the Contributor Guide unless I'm looking for a specific thing. I am easily left out of the loop having no overlap with other writers and sometimes these things don't get picked up by anyone during peer reviews so I just keep doing them until told otherwise.

**Comment by gsheldon** [comment]
Thanks for your review @sterobin !

File: docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc (hunk 5 above)
Comment (sterobin) [reply docs/product-assembly_case-management-getting-started/src/main/asciidoc/case-management-designing-a-case-proc.adoc:60]: @gsheldon, yeah, the time diff is killer on communications and other stuff. Sorry about that. There were several emails around the xref change and file/anchor naming though that I circulated back in about Sept/Oct of last year, one set of emails taking team input, and another confirming the decision. The only reason I remember it was that was about the time Marek told me not to expect too many writers to respond :) He said usually people don't say anything unless they care. But sometimes it could actually be missed, like in this case.

Either way, agreed that it's not worth your trouble to "fix" in retrospect. So long as it works, that's great.
